src/ir_Carrier.cpp

```diff
@@ -182,6 +192,100 @@ bool IRrecv::decodeCarrierAC40(decode_results *results, uint16_t offset,
+/// Alternative >64bit function to send CARRIER_AC48 messages
+/// Status: ALPHA / Untested.
+/// Where data is:
+///   uint8_t data[kCARRIER_AC48StateLength] = {0xA1, 0x20, 0x6B, 0xFF, 0xFF, 0xC5, 0x5E, 0xDF, 0x94, 0x00, 0x00, 0x3A};
+///
+/// @param[in] data An array of bytes containing the IR command.
+///                 It is assumed to be in MSB order for this code.
+/// @param[in] nbytes Nr. of bytes of data in the array. (>=kCARRIER_AC48StateLength)
+/// @param[in] repeat Nr. of times the message is to be repeated.
+void IRsend::sendCarrier_AC48(const uint8_t data[], const uint16_t nbytes, const uint16_t repeat) {
+  for (uint16_t r = 0; r <= repeat; r++) {
+    uint16_t pos = 0;
+    // Data Section #1
+    // e.g.
+    //   bits = 48; bytes = 6;
+    //   *(data + pos) = {0xA1, 0x20, 0x6B, 0xFF, 0xFF, 0xC5};
+    sendGeneric(kCarrier_AC48HdrMark, kCarrier_AC48HdrSpace,
+                kCarrier_AC48BitMark, kCarrier_AC48OneSpace,
+                kCarrier_AC48BitMark, kCarrier_AC48ZeroSpace,
+                kCarrier_AC48BitMark, kCarrier_AC48SpaceGap,
+                data + pos, 6,  // Bytes
+                kCarrier_AC48Freq, true, kNoRepeat, kDutyDefault);
+    pos += 6;  // Adjust by how many bytes of data we sent
+    // Data Section #2
+    // e.g.
+    //   bits = 48; bytes = 6;
+    //   *(data + pos) = {0x5E, 0xDF, 0x94, 0x00, 0x00, 0x3A};
+    sendGeneric(kCarrier_AC48HdrMark, kCarrier_AC48HdrSpace,
+                kCarrier_AC48BitMark, kCarrier_AC48OneSpace,
+                kCarrier_AC48BitMark, kCarrier_AC48ZeroSpace,
+                kCarrier_AC48BitMark, kDefaultMessageGap,
+                data + pos, 6,  // Bytes
+                kCarrier_AC48Freq, true, kNoRepeat, kDutyDefault);
+    pos += 6;  // Adjust by how many bytes of data we sent
+  }
+}
+#endif  // SEND_CARRIER_AC48
+
+#if DECODE_CARRIER_AC48
+/// Decode the supplied CARRIER_AC48 message.
+/// Function should be safe over 64 bits.
+/// Status: ALPHA / Untested.
+/// @param[in,out] results Ptr to the data to decode & where to store the decode
+/// @param[in] offset The starting index to use when attempting to decode the
+///   raw data. Typically/Defaults to kStartOffset.
+/// @param[in] nbits The number of data bits to expect.
+/// @param[in] strict Flag indicating if we should perform strict matching.
+/// @return A boolean. True if it can decode it, false if it can't.
+bool IRrecv::decodeCarrier_AC48(decode_results *results, uint16_t offset, const uint16_t nbits, const bool strict) {
+  if (results->rawlen < 2 * nbits + kCarrier_AC48Overhead - offset)
+    return false;  // Too short a message to match.
+  if (strict && nbits != kCarrierAc48Bits)
+    return false;
+
+  uint16_t pos = 0;
+  uint16_t used = 0;
+
+  // Data Section #1
+  // e.g.
+  //   bits = 48; bytes = 6;
+  //   *(results->state + pos) = {0xA1, 0x20, 0x6B, 0xFF, 0xFF, 0xC5};
+  used = matchGeneric(results->rawbuf + offset, results->state + pos,
```

Comment:
Same here, do we want this to be similar to `decodeCarrierAC ?

Reply:
Either will work, but we should probably go with the Integer version.